### PR TITLE
Chief Engineer Vendor Update

### PR DESCRIPTION
**Pull request by morrowwolf**
## About The Pull Request

Adds more options to chief engineer vendor as well as extending the deployment section.

## Why It's Good For The Game

More customization for looks is good. Chief Engineer is tasked with deploying to assist with FOB fairly often and giving basic options will lessen his time asking for a dozen things at requisition.

## Changelog

:cl: Morrow
add: More options to chief engineer vendor as well as an extended section for deployment gear.
/:cl:

### CODE REVIEW

File: code/modules/cm_marines/equipment/guncases.dm [hunk 1]
```diff
@@ -157,3 +157,28 @@
...
+	desc = "A gun case containing the M41A pulse rifle MK2."
+	storage_slots = 5
+	can_hold = list(/obj/item/weapon/gun/rifle/m41aMK1, /obj/item/ammo_magazine/rifle/m41aMK1)
+
+/obj/item/storage/box/guncase/m41a/fill_preset_inventory()
+	new /obj/item/weapon/gun/rifle/m41a(src)
+	new /obj/item/ammo_magazine/rifle(src)
+	new /obj/item/ammo_magazine/rifle(src)
+	new /obj/item/ammo_magazine/rifle(src)
+	new /obj/item/ammo_magazine/rifle(src)
```
Comment (stanalbatross) [inline code/modules/cm_marines/equipment/guncases.dm:172]: use var/i 1 to 4

File: code/modules/cm_marines/equipment/guncases.dm [hunk 2]
```diff
@@ -157,3 +157,28 @@
 	new /obj/item/ammo_magazine/m2c(src)
 	new /obj/item/storage/belt/marine/m2c(src)
 
+//------------
+/obj/item/storage/box/guncase/m41a
+	name = "\improper M41A pulse rifle MK2 case"
+	desc = "A gun case containing the M41A pulse rifle MK2."
+	storage_slots = 5
+	can_hold = list(/obj/item/weapon/gun/rifle/m41aMK1, /obj/item/ammo_magazine/rifle/m41aMK1)
```
Comment (stanalbatross) [inline code/modules/cm_marines/equipment/guncases.dm:165]: ?

File: code/game/machinery/vending/vendor_types/crew/senior_officers.dm [hunk 3]
```diff
@@ -70,46 +70,86 @@ GLOBAL_LIST_INIT(cm_vending_clothing_military_police_chief, list(
...
+		list("Fuel Tank Strap Pouch", 0, /obj/item/storage/pouch/flamertank, (MARINE_CAN_BUY_COMBAT_R_POUCH|MARINE_CAN_BUY_COMBAT_L_POUCH), VENDOR_ITEM_REGULAR),
+		
+		list("MASK (CHOOSE 1)", 0, null, null, null),
+		list("Gas Mask", 0, /obj/item/clothing/mask/gas, MARINE_CAN_BUY_MASK, VENDOR_ITEM_REGULAR),
+		list("Heat Absorbent Coif", 0, /obj/item/clothing/mask/rebreather/scarf, MARINE_CAN_BUY_MASK, VENDOR_ITEM_REGULAR),
+		
+		list("PRIMARY FIREARMS (CHOOSE 1)", 0, null, null, null),
+		list("M37A2 Pump Shotgun", 0, /obj/item/storage/box/guncase/pumpshotgun, MARINE_CAN_BUY_ESSENTIALS, VENDOR_ITEM_REGULAR),
+		list("M41A Pulse Rifle MK2", 0, /obj/item/storage/box/guncase/m41a, MARINE_CAN_BUY_ESSENTIALS, VENDOR_ITEM_REGULAR),
+		list("M240 Incinerator Unit", 0, /obj/item/storage/box/guncase/flamer, MARINE_CAN_BUY_ESSENTIALS, VENDOR_ITEM_REGULAR),
```
Comment (stanalbatross) [inline code/game/machinery/vending/vendor_types/crew/senior_officers.dm:151]: this is much stronger than the other 2 btw

File: code/modules/cm_marines/equipment/guncases.dm [hunk 4]
```diff
@@ -157,3 +157,28 @@
...
+/obj/item/storage/box/guncase/pumpshotgun
+	name = "\improper M37A2 Pump Shotgun case"
+	desc = "A gun case containing the M37A2 Pump Shotgun."
+	storage_slots = 3
+	can_hold = list(/obj/item/weapon/gun/shotgun/pump, /obj/item/ammo_magazine/shotgun/buckshot)
+
+/obj/item/storage/box/guncase/pumpshotgun/fill_preset_inventory()
+	new /obj/item/weapon/gun/shotgun/pump(src)
+	new /obj/item/ammo_magazine/shotgun/buckshot(src)
+	new /obj/item/ammo_magazine/shotgun/buckshot(src)
```
Comment (stanalbatross) [inline code/modules/cm_marines/equipment/guncases.dm:184]: var/i 1 to 2 also maybe make the ammo type random/more varied?

File: code/game/machinery/vending/vendor_types/crew/senior_officers.dm [hunk 5]
```diff
@@ -70,46 +70,86 @@ GLOBAL_LIST_INIT(cm_vending_clothing_military_police_chief, list(
...
+		list("M4A3 Custom Pistol", 0, /obj/item/storage/belt/gun/m4a3/commander, MARINE_CAN_BUY_SECONDARY, VENDOR_ITEM_RECOMMENDED),
+		list("Mod 88 Pistol", 0, /obj/item/storage/belt/gun/m4a3/mod88, MARINE_CAN_BUY_SECONDARY, VENDOR_ITEM_RECOMMENDED),
+		list("M44 Revolver", 0, /obj/item/storage/belt/gun/m44/mp, MARINE_CAN_BUY_SECONDARY, VENDOR_ITEM_RECOMMENDED),
+		
 		list("ACCESSORIES (CHOOSE 1)", 0, null, null, null),
 		list("Black Webbing Vest", 0, /obj/item/clothing/accessory/storage/black_vest, MARINE_CAN_BUY_ACCESSORY, VENDOR_ITEM_REGULAR),
 		list("Brown Webbing Vest", 0, /obj/item/clothing/accessory/storage/black_vest/brown_vest, MARINE_CAN_BUY_ACCESSORY, VENDOR_ITEM_RECOMMENDED),
 		list("Webbing", 0, /obj/item/clothing/accessory/storage/webbing, MARINE_CAN_BUY_ACCESSORY, VENDOR_ITEM_REGULAR),
-
+		list("Shoulder Holster", 0, /obj/item/clothing/accessory/storage/holster, MARINE_CAN_BUY_ACCESSORY, VENDOR_ITEM_REGULAR),
```
Comment (stanalbatross) [inline code/game/machinery/vending/vendor_types/crew/senior_officers.dm:124]: needs drop pouch

File: code/game/machinery/vending/vendor_types/crew/senior_officers.dm [hunk 6]
```diff
@@ -70,46 +70,86 @@ GLOBAL_LIST_INIT(cm_vending_clothing_military_police_chief, list(
...
+		list("Hardhat", 0, /obj/item/clothing/head/hardhat/white, MARINE_CAN_BUY_HELMET, VENDOR_ITEM_REGULAR),
+		list("Patrol Cap", 0, /obj/item/clothing/head/cmcap, MARINE_CAN_BUY_HELMET, VENDOR_ITEM_REGULAR),
+		list("Welding Helmet", 0, /obj/item/clothing/head/welding, MARINE_CAN_BUY_HELMET, VENDOR_ITEM_REGULAR),
+		list("Beret, Engineering", 0, /obj/item/clothing/head/beret/eng, MARINE_CAN_BUY_HELMET, VENDOR_ITEM_REGULAR),
+		
+		list("SUIT (CHOOSE 1)", 0, null, null, null),
+		list("Orange Hazard Vest", 0, /obj/item/clothing/suit/storage/hazardvest, MARINE_CAN_BUY_ARMOR, VENDOR_ITEM_REGULAR),
+		list("Blue Hazard Vest", 0, /obj/item/clothing/suit/storage/hazardvest/blue, MARINE_CAN_BUY_ARMOR, VENDOR_ITEM_REGULAR),
+		list("Yellow Hazard Vest", 0, /obj/item/clothing/suit/storage/hazardvest/yellow, MARINE_CAN_BUY_ARMOR, VENDOR_ITEM_REGULAR),
+		list("Black Hazard Vest", 0, /obj/item/clothing/suit/storage/hazardvest/black, MARINE_CAN_BUY_ARMOR, VENDOR_ITEM_REGULAR),
```
Comment (stanalbatross) [inline code/game/machinery/vending/vendor_types/crew/senior_officers.dm:96]: alphabetical order

File: code/game/machinery/vending/vendor_types/crew/senior_officers.dm [hunk 7]
```diff
@@ -70,46 +70,86 @@ GLOBAL_LIST_INIT(cm_vending_clothing_military_police_chief, list(
...
+		
+		list("UNIFORM (CHOOSE 1)", 0, null, null, null),
+		list("Chief Engineer Uniform", 0, /obj/item/clothing/under/marine/officer/ce, MARINE_CAN_BUY_UNIFORM, VENDOR_ITEM_RECOMMENDED),
+		list("Service Uniform", 0, /obj/item/clothing/under/marine/officer/bridge, MARINE_CAN_BUY_UNIFORM, VENDOR_ITEM_REGULAR),
+		
+		list("HELMET (CHOOSE 1)", 0, null, null, null),
+		list("Hardhat", 0, /obj/item/clothing/head/hardhat/white, MARINE_CAN_BUY_HELMET, VENDOR_ITEM_REGULAR),
+		list("Patrol Cap", 0, /obj/item/clothing/head/cmcap, MARINE_CAN_BUY_HELMET, VENDOR_ITEM_REGULAR),
+		list("Welding Helmet", 0, /obj/item/clothing/head/welding, MARINE_CAN_BUY_HELMET, VENDOR_ITEM_REGULAR),
+		list("Beret, Engineering", 0, /obj/item/clothing/head/beret/eng, MARINE_CAN_BUY_HELMET, VENDOR_ITEM_REGULAR),
```
Comment (stanalbatross) [inline code/game/machinery/vending/vendor_types/crew/senior_officers.dm:90]: alphabetical

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/game/machinery/vending/vendor_types/crew/senior_officers.dm (hunk 3 above)
Comment (morrowwolf) [reply code/game/machinery/vending/vendor_types/crew/senior_officers.dm:151]: Flamer is real good for clearing weeds from safety and the CE shouldn't be contributing to sieges/hunting anyways. Should hopefully be a non-factor. I will remove it if you ask though.

File: code/modules/cm_marines/equipment/guncases.dm [hunk 8]
```diff
@@ -157,3 +157,34 @@
...
+/obj/item/storage/box/guncase/pumpshotgun
+	name = "\improper M37A2 Pump Shotgun case"
+	desc = "A gun case containing the M37A2 Pump Shotgun."
+	storage_slots = 4
+	can_hold = list(/obj/item/weapon/gun/shotgun/pump, /obj/item/ammo_magazine/shotgun/buckshot, /obj/item/ammo_magazine/shotgun/flechette, /obj/item/ammo_magazine/shotgun/slugs)
+
+/obj/item/storage/box/guncase/pumpshotgun/fill_preset_inventory()
+	new /obj/item/weapon/gun/shotgun/pump(src)
+	for(var/i = 1 to 3)
+		var/random_pick = pick(1, 2, 3)
```
Comment (stanalbatross) [inline code/modules/cm_marines/equipment/guncases.dm:183]:
```suggestion
		var/random_pick = rand(1, 3)
```

**Review by stanalbatross — APPROVED** [review]

**Comment by stanalbatross** [comment]
if the flamer is powergamed or abused it can be swapped out later

**Comment by nuklearcellphoneg** [comment]
🧎‍♂️

**Pull request merged**